src/path.cr

```diff
@@ -710,8 +710,8 @@ struct Path
     if home
       if name == "~"
         name = resolve_home(home).to_s
-      elsif name.starts_with?("~/")
-        name = resolve_home(home).join(name.byte_slice(2, name.bytesize - 2)).to_s
+      elsif sep = Path.separators(@kind).find { |sep| name.starts_with?("~#{sep}") }
```

Comment:
I think it's fine to hard code the home expansion indicators. There are only two options. This avoids unnecessary string allocations.

```suggestion
      elsif name.starts_with?("~/") || (@kind.windows? && name.starts_with?("~\\"))
```

We could technically memoize the expanded comparison strings, but that's hardly necessary.

Reply:
Alternatively, we could check if the first character is `~` and the following matches a separator. But again, I think hard coding is totally fine here.